datafusion/common/src/scalar.rs

```diff
@@ -2297,14 +2326,20 @@ impl ScalarValue {
                 Self::Dictionary(key_type.clone(), Box::new(value))
             }
             DataType::Struct(fields) => {
-                let array = as_struct_array(array)?;
+                let struct_arr = as_struct_array(array)?;
```

Comment:
I think this could be implemented using https://docs.rs/arrow/latest/arrow/array/trait.Array.html#tymethod.slice

So like 
```rust
let one_element_struct_array = array.slice(index, 1)
  .as_struct_array()?
```

Which would be both less code as well as more efficient (no data copies needed)